src/main/scala/codacy/plugins/test/multiple/MultipleTests.scala

```diff
@@ -0,0 +1,90 @@
+package codacy.plugins.test.multiple
+
+import codacy.plugins.test._
+
+import com.codacy.analysis.core.model.{CodacyCfg, Configuration, FileCfg, FileError, FullLocation, Issue, LineLocation}
+import com.codacy.analysis.core.tools.Tool
+import com.codacy.plugins.api.results.Result
+import com.codacy.plugins.results.traits.{DockerToolDocumentation, ToolRunner}
+import com.codacy.plugins.runners.{BinaryDockerRunner, DockerRunner}
+import com.codacy.plugins.utils.BinaryDockerHelper
+import com.codacy.plugins.results.PluginResult
+
+import better.files._
+import java.io.{ File => JFile }
+
+import scala.util.{Failure, Success, Try}
+import scala.xml.XML
+import com.codacy.analysis.core.model.ToolResult
+import com.codacy.analysis.core.model.Location
+import scala.util.Properties
+
+object MultipleTests extends ITest {
+
+  val opt = "multiple"
+
+  def run(docsDirectory: JFile, dockerImage: DockerImage, optArgs: Seq[String]): Boolean = {
+    debug(s"Running MultipleTests:")
+    val multipleTestsDirectory = docsDirectory.toScala / DockerHelpers.multipleTestsDirectoryName
+    val languages = findLanguages(multipleTestsDirectory.toJava, dockerImage)
+    val dockerTool = createDockerTool(languages, dockerImage)
+    val toolDocumentation = new DockerToolDocumentation(dockerTool, new BinaryDockerHelper(useCachedDocs = false))
+    val dockerRunner = new BinaryDockerRunner[Result](dockerTool)()
+    val runner = new ToolRunner(dockerTool, toolDocumentation, dockerRunner)
+    val tools = languages.map(new Tool(runner, DockerRunner.defaultRunTimeout)(dockerTool, _))
+    multipleTestsDirectory.list.forall { testDirectory =>
+      val srcDir = testDirectory / "src"
+      val resultFile = testDirectory / "results.xml"
+      val resultFileXML = XML.loadFile(resultFile.toJava)
+      val expectedResults = CheckstyleFormatParser.parseResultsXml(resultFileXML).toSet
+      debug(s"${testDirectory.name} should have ${expectedResults.size} results")
+      val configuration = createConfiguration(testDirectory, srcDir)
+      tools.exists { tool =>
+        val res = runTool(tool, srcDir, configuration)
+        ResultPrinter.printToolResults(res, expectedResults)
+      }
+    }
+  }
+
+  private def createConfiguration(testDirectory: File, srcDir: File) = {
+    val patternsPath = testDirectory / "patterns.xml"
+    if (patternsPath.exists) {
+      val patternsFileXML = XML.loadFile(patternsPath.toJava)
+      val (patterns, extraValues) = CheckstyleFormatParser.parsePatternsXml(patternsFileXML)
+      if (patterns.isEmpty) FileCfg(Some(srcDir.pathAsString), extraValues)
+      else CodacyCfg(patterns, Some(srcDir.pathAsString), extraValues)
+    } else FileCfg(Some(srcDir.pathAsString), None)
+  }
+
+  private def convertResults(resultSet: Set[ToolResult]): Set[Either[String, PluginResult]] = resultSet.map {
+    case Issue(patternId, filename, message, level, category, location: Location) =>
+      val line = location match {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Default case not specified.](https://app.codacy.com/app/Codacy/codacy-plugins-test/pullRequest?prid=4500290)